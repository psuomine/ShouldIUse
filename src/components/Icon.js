import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Container = styled.div`
  display: flex;
  width: 48px;
  height: 48px;
  background-color: ${({ backgroundColor }) => backgroundColor};
  border-radius: 100px;
`

export default class Icon extends Component {
  render() {
    return (
      <Container type={this.props.backgroundColor}>
        <div />
      </Container>
    )
  }
}

Icon.propTypes = {
  backgroundColor: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['success', 'error']),
}
