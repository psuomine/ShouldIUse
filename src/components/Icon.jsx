import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const IconSvg = styled.svg`
  fill: ${({ color, theme }) => color || theme.primary};
`

const Icon = ({ icon, height, width, color }) => (
  <IconSvg data-testid="svg-icon" color={color} width={width} height={height}>
    <path data-testid="svg-icon-path" d={icon} />
  </IconSvg>
)

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
  height: PropTypes.number,
  width: PropTypes.number,
  color: PropTypes.string,
}

Icon.defaultProps = {
  height: 24,
  width: 24,
  color: '',
}

export default Icon
