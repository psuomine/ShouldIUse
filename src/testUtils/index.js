import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import Theme from 'theme/Theme'

const customRender = (node, options) => render(<ThemeProvider theme={Theme}>{node}</ThemeProvider>, options)

// re-export everything
export * from 'react-testing-library'

// override render method
export { customRender as render }
