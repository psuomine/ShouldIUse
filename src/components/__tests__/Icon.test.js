import React from 'react'
import { render } from 'testUtils'
import Icon from '../Icon'

const iconPath = 'M7.42449 14.5737L2.42002'

describe('<Icon />', () => {
  it('renders icon based on props', () => {
    const { getByTestId } = render(<Icon icon={iconPath} />)
    const icon = getByTestId('svg-icon')
    const svgPath = getByTestId('svg-icon-path')
    expect(svgPath).toHaveAttribute('d')
    expect(icon.firstChild).toMatchSnapshot()
  })
})
