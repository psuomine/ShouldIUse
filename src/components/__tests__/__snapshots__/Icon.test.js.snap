// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders icon based on props 1`] = `
<path
  d="M7.42449 14.5737L2.42002"
  data-testid="svg-icon-path"
/>
`;
